src/Wave.cpp

```diff
@@ -0,0 +1,147 @@
+    parameters.length_overall = utilities::getParameter<double>("Wave", el, "length_overall", "m", 0);
+    parameters.water_density = utilities::getParameter<double>("Wave", el, "water_density", "kg/m3", 1.12);
+    parameters.coefficients = utilities::getParameter<Vector3d>("Wave", el, "wave_coeffs",
+                                                                "", Vector3d(0, 0, 0));
+    return parameters;
+}
+
+void Wave::readWaveVelocity(const ConstVector3dPtr &velocity)
+{
+    mWaveVelocity = Vector3d(velocity->x(), velocity->y(), velocity->z());
+}
+
+void Wave::readWaveFrequency(const ConstVector3dPtr &frequency)
+{
+    mWaveFrequency = Vector3d(frequency->x(), frequency->y(), frequency->z());
+}
+
+Wave::Effects Wave::computeEffects(Quaterniond const body2world_orientation, Vector3d const vessel_linear_vel_world, Vector3d const wave_velocity_world, Vector3d const wave_frequency_world) const
+{
+    Quaterniond world2body_orientation = body2world_orientation.Inverse();
+    Vector3d relative_wave_velocity_world = vessel_linear_vel_world - wave_velocity_world;
+    Vector3d relative_wave_velocity_body = world2body_orientation * relative_wave_velocity_world;
+    relative_wave_velocity_body.Z() = 0;
+    if (relative_wave_velocity_body.Length() < 1e-3)
+        return Effects{};
+
+    // Compute the wave's angle of attack and its coefficients
+    double angle_of_attack_rad = -atan2(relative_wave_velocity_body.Y(), relative_wave_velocity_body.X());
+    Angle angle_of_attack(angle_of_attack_rad);
+    double wave_coeff_x = -mParameters.coefficients.X() * cos(angle_of_attack.Radian());
+    double wave_coeff_y = mParameters.coefficients.Y() * sin(angle_of_attack.Radian());
+    double wave_coeff_z = mParameters.coefficients.Z();
+    double wave_coeff_n = mParameters.torque_constant *sin(2 * angle_of_attack.Radian());
+
+    // Compute the wave's time coefficients
+    time_t seconds = time(NULL);
+    double wave_coeff_time_x = sin(6.28*seconds*wave_frequency_world.X());
+    double wave_coeff_time_y = sin(6.28*seconds*wave_frequency_world.Y());
+    double wave_coeff_time_z = sin(6.28*seconds*wave_frequency_world.Z());
+
+    // Compute wave effects for X, Y and N
+    Effects wave_effects;
+    wave_effects.force[0] = 0.5 * wave_coeff_time_x * mParameters.water_density * relative_wave_velocity_body.SquaredLength() * wave_coeff_x * mParameters.frontal_area;
+    wave_effects.force[1] = 0.5 * wave_coeff_time_y * mParameters.water_density * relative_wave_velocity_body.SquaredLength() * wave_coeff_y * mParameters.lateral_area;
+    wave_effects.force[2] = 0.5 * wave_coeff_time_z * mParameters.water_density * relative_wave_velocity_body.SquaredLength() * wave_coeff_z * mParameters.bottom_area;
+    wave_effects.torque[2] = 0.5 * mParameters.water_density * relative_wave_velocity_body.SquaredLength() * wave_coeff_n * mParameters.lateral_area * mParameters.length_overall;
```

Comment:
These seem awfully close to the wind effect calculations. Are you sure this is the way to calculate effects for waves?

Reply:
It is most definitely not, but we decided to let it slide.

However, @kapeps it would make awful sense to put the sinusoid in the plugin instead of externalizing it within Syskit, no ?

Reply:
I think now it is, probably it wasnt by the time @doudou commented. I dont know why you kept the coeffs though, wouldnt the wave effects be using the sinusoidal directly at the link for now?